common/src/main/java/com/wynntils/mc/mixin/PlayerMixin.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.mc.mixin;
+
+import com.wynntils.mc.EventFactory;
+import com.wynntils.mc.event.ArmSwingEvent;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.entity.player.Player;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+@Mixin(Player.class)
+public abstract class PlayerMixin {
+
+    @Redirect(
+            method = "drop(Lnet/minecraft/world/item/ItemStack;ZZ)Lnet/minecraft/world/entity/item/ItemEntity;",
+            at =
+                    @At(
+                            value = "INVOKE",
+                            target =
+                                    "Lnet/minecraft/world/entity/player/Player;swing(Lnet/minecraft/world/InteractionHand;)V"))
+    private void onSwingInInventoryScreen(Player player, InteractionHand hand) {
+        if (!EventFactory.onArmSwing(ArmSwingEvent.ArmSwingContext.DROP_ITEM_FROM_INVENTORY_SCREEN, hand)
```

Comment:
Please flip this if condition. Return if event is cancelled, otherwise swing (to reduce code right shift)